materials_io/base.py

```diff
@@ -59,41 +58,57 @@ def parse(self, group: List[str], context: dict = None) -> dict:
             (dict): The parsed results, in JSON-serializable format.
         """
 
-    def parse_as_unit(self, files: List[str]) -> dict:
+    def parse_as_unit(self, files, context: dict = None) -> dict:
```

Comment:
Could you put my type annotation back? Perhaps use `Union[str, List[str]]`?

I am experimenting with being diligent about typing, to see what kind of benefits we'd reap later. For now, pycharm is giving me some helpful feedback on type match ups.

Reply:
I'm somewhat cautious about the type annotations, because it's easy for those to become misaligned or too narrow (`List`, for example, is not required, because any `Iterable` will do). But I do understand the benefits, so I'll add them back.

Reply:
Yea, choosing between `List` vs `Tuple` vs `Iterator` did confuse me a bit. Maybe I'll figure it out with more practice.